conans/tools.py

```diff
@@ -165,6 +165,17 @@ def build_sln_command(settings, sln_path, targets=None, upgrade_project=True, bu
         command += " /target:%s" % ";".join(targets)
     return command
 
+def vs_installation_path(version):
+    if not hasattr(vs_installation_path, "_cached"):
+        vs_installation_path._cached = dict()
+    if not version in vs_installation_path._cached.keys():
```

Comment:
keys() unnecesary here. ``var in dict()`` is a valid python expression that looks if var is in keys of dict